src/main/java/com/microsoft/jfr/RecordingOptions.java

```diff
@@ -0,0 +1,385 @@
+            options.put(
+                    Option.DISK, 
+                    Boolean.valueOf(disk).toString()
+            );
+            return this;
+        }
+
+        /**
+         * Sets how long the recording should be running. The default is to have no limit.
+         * If the value is "0", no limit is imposed. Otherwise, the value is the string
+         * representation of a positive long value followed by an empty space and one
+         * of the following units:
+         * <ul>
+         *     <li>"ns" (nanoseconds)</li>
+         *     <li>"us" (microseconds)</li>
+         *     <li>"ms" (milliseconds)</li>
+         *     <li>"s" (seconds)</li>
+         *     <li>"m" (minutes)</li>
+         *     <li>"h" (hours)</li>
+         *     <li>"d" (days)</li>
+         * </ul>
+         * For example, {@code "2 h"}, {@code "3 m"}.
+         *
+         * If the value is {@code null}, {@code duration} will be set to the default value,
+         * which is "no limit".
+         *
+         * @param duration The maximum length of time a recording should be running.
+         * @return {@code this}
+         * @throws IllegalArgumentException The {@code duration} parameter is not formatted correctly,
+         * or represents a negative value
+         */
+        public Builder duration(String duration) throws IllegalArgumentException {
+            options.put(
+                    Option.DURATION,
+                    validateDuration(Option.DURATION, duration)
+            );
+            return this;
+        }
+
+        /**
+         * Construct a {@code Configuration} from the options that were set on this builder.
+         * The {@code Builder} state is not reset by calling this method.
+         * @return A {@code Configuration}, never {@code null}.
+         */
+        public RecordingOptions build() {
+            return new RecordingOptions(this);
+        }
+
+    }
+
+    /**
+     * Constructor is private. Only the Builder can construct a Configuration.
+     * @param builder The builder that was used to parameterize the configuration
+     */
+    private RecordingOptions(Builder builder) {
+        // Note we're converting from Map<Option<?>,String> to Map<String,String>
+        final Map<Option,String> options = builder.options;
+        //
+        // For each option,
+        //    if the option is not the default value
+        //    add the option name and value to recordingOptions
+        // An option is the default value if
+        //     it was not set in the builder,
+        //     it was not set as a system property,
+        //     or was set in the builder, but to a default value (builder.setDuration(null), for example)
+        // This stream does not modify builder.options.
+        Map<String,String> initRecordingOptions =
+                Stream.of(Option.values())
+                        .filter(option -> !option.defaultValue.equals(options.getOrDefault(option, option.defaultValue)))
```

Comment:
what is this really saying?  Maybe extract for clarity?  OK it's in the comments